### PR TITLE
Port IOApp.Simple to CE2

### LINKED ISSUE #1721: Port IOApp.Simple to CE2
In series/3.x, a simpler interface for IOApp is provided, for the users who don't need args or exit codes:

```scala
object Main extends IOApp.Simple:
  def run: IO[Unit] = IO.println("hello")
```

I think it could be backported to CE2 for the poor souls who won't be able to migrate for a while for reasons.

### PR DESCRIPTION
**Pull request by alexandrustana**
This PR  fixes #1721 by porting the `Simple` trait  from CE3 to CE2

### CODE REVIEW

**Review by vasilmkd — APPROVED** [review]
Nice. Thank you.

**Comment by djspiewak** [comment]
Very nice! Thank you!

File: benchmarks/shared/src/main/scala/cats/effect/benchmarks/ECBenchmark.scala [hunk 1]
```diff
@@ -25,31 +25,30 @@ import scala.concurrent.ExecutionContext
...
+  private val ioApp = new IOApp.Simple with Run
   private val ioAppCtx = new IOApp.WithContext with Run {
     protected def executionContextResource: Resource[SyncIO, ExecutionContext] =
       Resource.eval(SyncIO.pure(ExecutionContext.Implicits.global))
   }
 
   @Benchmark
   def app(): Unit = {
-    val _ = ioApp.main(Array.empty)
+    val _ = ioApp.run
```
Comment (kubukoz) [inline benchmarks/shared/src/main/scala/cats/effect/benchmarks/ECBenchmark.scala:47]: I *think* this change makes the benchmarks not actually run the app at all 😅

Comment (kubukoz) [reply benchmarks/shared/src/main/scala/cats/effect/benchmarks/ECBenchmark.scala:47]: Following up in https://github.com/typelevel/cats-effect/pull/1769

**Pull request merged**